apps/web/screens/UniversitySearch/UniversitySearch.tsx

```diff
@@ -151,6 +156,8 @@ const UniversitySearch: Screen<UniversitySearchProps> = ({
   const [totalPages, setTotalPages] = useState<number>(0)
   const [filters, setFilters] = useState<FilterProps>(filtersFromQuery)
   const titleRef = useRef<HTMLDivElement>(null)
+  const [searchTimeoutId, setSearchTimeoutId] =
```

Comment:
I wonder if it makes more sense for this to be a ref instead 🤔

Reply:
Yeah possibly, Ill consider re-factoring on a second pass, might reduce some re-renders using ref instead and get rid of async-ness of setting state